### ISSUE TITLE
cy.server whitelist behaves as blacklist

### ISSUE THREAD

**Issue by alex-golubtsov** [cypress-io/cypress#6642]
### Current behavior:

Currently `whitelist` option for `cy.server()` actually works as blacklist due to this line
https://github.com/cypress-io/cypress/blob/master/packages/driver/src/cypress/server.coffee#L236

In other words:
```js
cy.server({ whitelist: () => true })
```
will ignore any request

and 
```js
cy.server({ whitelist: () => false})
```
will catch all requests
### Desired behavior:

Exactly opposite
```js
cy.server({ whitelist: () => true })
```
should catch all requests

and 
```js
cy.server({ whitelist: () => false})
```
should ignore any request

### Test code to reproduce

```js
cy.server()
cy.route(<any route>)
```

### Versions
`4.1.0`, also reproducible in `master`
win10, chrome

**Comment by jennifer-shehane** [comment]
yeeeaaah...I can see this. This would require a breaking change.

**Comment by jennifer-shehane** [comment]
Released in [5.0.0](https://on.cypress.io/changelog#5-0-0).

This comment thread has been locked. If you are still experiencing this issue after upgrading to Cypress v5.0.0, please open a new issue.
